src/main/java/uk/gov/hmcts/darts/task/service/impl/ManualDeletionProcessorImpl.java

```diff
@@ -83,13 +97,48 @@ private void deleteTranscriptionDocument(UserAccountEntity userAccount, Transcri
             deleteFromExternalDataStore(externalObjectDirectoryEntity);
             externalObjectDirectoryRepository.delete(externalObjectDirectoryEntity);
         }
-
         transcription.markAsDeleted(userAccount);
         transcriptionDocumentRepository.save(transcription);
+        processArmEods(transcription.getDeletedTs(),
+                       objectAdminAction,
+                       externalObjectDirectoryRepository.findByTranscriptionDocumentEntityAndExternalLocationTypeAndStatus(transcription,
+                                                                                                                                 EodHelper.armLocation(),
+                                                                                                                                 EodHelper.storedStatus()));
         logApi.transcriptionDeleted(transcription.getId());
     }
 
-    private void deleteFromExternalDataStore(ExternalObjectDirectoryEntity externalObjectDirectoryEntity) {
+
+    void processArmEods(OffsetDateTime deletedTs, ObjectAdminActionEntity objectAdminAction,
+                        List<ExternalObjectDirectoryEntity> externalObjectDirectoryEntities) {
+        externalObjectDirectoryEntities
+            .forEach(externalObjectDirectoryEntity -> processArmEod(deletedTs, objectAdminAction, externalObjectDirectoryEntity));
+    }
+
+    void processArmEod(OffsetDateTime deletedTs, ObjectAdminActionEntity objectAdminAction,
+                        ExternalObjectDirectoryEntity externalObjectDirectoryEntity) {
+        armDataManagementApi.updateMetadata(
+            externalObjectDirectoryEntity.getExternalRecordId(),
+            deletedTs.minusYears(eventDateAdjustmentYears),
+            getRetConfReason(deletedTs, objectAdminAction)
+        );
+        externalObjectDirectoryRepository.delete(externalObjectDirectoryEntity);
+    }
+
+    String getRetConfReason(OffsetDateTime deletedTs, ObjectAdminActionEntity objectAdminAction) {
+        try {
+            RetConfReason retConfReason = new RetConfReason();
+            retConfReason.manualDeletionTs = deletedTs;
+            retConfReason.manualDeletionReason = objectAdminAction.getObjectHiddenReason().getReason();
+            retConfReason.ticketReference = objectAdminAction.getTicketReference();
+            retConfReason.comments = objectAdminAction.getComments();
+            return objectMapper.writeValueAsString(retConfReason);
+        } catch (Exception e) {
+            log.error("Error while creating RetConfReason", e);
+            return "Error while creating RetConfReason";
```

Comment:
If there is an error here should this error be sent to ARM? Isn't this an internal issue?

Reply:
We should not pass any random error message to ARM if the agreed contract is a JSON.
The entire process should be Transactional i.e. If it fails in any of the step, the entire manual deletion process should fail for that object. This way the isDeleted should stay FALSE and will be picked up in the next execution. 
The update to Media isDeleted should happen at the end.

Reply:
I have updated this code to chuck an exception so any existing error handling logic can kick in

Reply:
Does the existing error handling rollback the transaction? If the ARM call fails for any reason, the isdeleted flag in Media should be rollback so that it gets picked up again in next iteration.

Reply:
Looks like there was no exception handling in this class before I have added one and added some transactions to ensure this gets handled correctly